chia/wallet/util/puzzle_representation.py

```diff
@@ -0,0 +1,72 @@
+from typing import Dict, List, Union
+
+from chia.types.blockchain_format.program import Program
+from chia.types.blockchain_format.sized_bytes import bytes32
+
+from chia.util.ints import uint32
+
+
+class PuzzleRepresentation:
+    """
+    This class is intended to be vehicle for compression and serialization of an arbitrary configuration of
+    "outer" and "inner" puzzles.  This can act as a generalized "driver" for an arbitrary puzle.
+    The idea is that you make a complex stack of puzzles, represent it with this class, and then you don't have
+    to worry about writing a whole new driver file for your unique stack, you can instead rely on the component
+    drivers to work together as if it were a driver for the whole puzzle.
+    """
+
+    base: bytes32
+    args: List[Union["PuzzleRepresentation", Program]]
+
+    def __init__(self, base, args):
+        self.base = base
+        self.args = args
```

Comment:
Delete this and use `@dataclass`.  Generally, every class should be one.

Reply:
What is `args` for? I'm confused by this vague name.